providers/sfmesh/sfmesh.go

```diff
@@ -0,0 +1,850 @@
+package sfmesh
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/preview/preview/servicefabricmesh/mgmt/servicefabricmesh"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/virtual-kubelet/virtual-kubelet/manager"
+	"github.com/virtual-kubelet/virtual-kubelet/providers"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const (
+	defaultCPUCapacity    = "60"
+	defaultMemoryCapacity = "48Gi"
+	defaultPodCapacity    = "5"
+	defaultCPURequests    = 1.0
+	defaultMemoryRequests = 1.0
+	defaultCPULimit       = 4.0
+	defaultMemoryLimit    = 16.0
+)
+
+// SFMeshProvider implements the Virtual Kubelet provider interface
+type SFMeshProvider struct {
+	nodeName           string
+	operatingSystem    string
+	internalIP         string
+	daemonEndpointPort int32
+	appClient          *servicefabricmesh.ApplicationClient
+	networkClient      *servicefabricmesh.NetworkClient
+	serviceClient      *servicefabricmesh.ServiceClient
+	region             string
+	resourceGroup      string
+	resourceManager    *manager.ResourceManager
+}
+
+// AuthConfig is the secret returned from an ImageRegistryCredential
+type AuthConfig struct {
+	Username      string `json:"username,omitempty"`
+	Password      string `json:"password,omitempty"`
+	Auth          string `json:"auth,omitempty"`
+	Email         string `json:"email,omitempty"`
+	ServerAddress string `json:"serveraddress,omitempty"`
+	IdentityToken string `json:"identitytoken,omitempty"`
+	RegistryToken string `json:"registrytoken,omitempty"`
+}
+
+// NewSFMeshProvider creates a new SFMeshProvider
+func NewSFMeshProvider(rm *manager.ResourceManager, nodeName, operatingSystem string, internalIP string, daemonEndpointPort int32) (*SFMeshProvider, error) {
+	azureSubscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	azureTenantID := os.Getenv("AZURE_TENANT_ID")
+	azureClientID := os.Getenv("AZURE_CLIENT_ID")
+	azureClientSecret := os.Getenv("AZURE_CLIENT_SECRET")
+	region := os.Getenv("REGION")
+	resourceGroup := os.Getenv("RESOURCE_GROUP")
+
+	if azureSubscriptionID == "" {
```

Comment:
wondering whether you want to integrate with AKS/ACS tightly to load the /etcu/kubernetes/azure.json from the host, so that the user doesn't need to specify any of these settings but uses the AKS/ACS one.

https://github.com/virtual-kubelet/virtual-kubelet/blob/3f83588e599075aee7958620f62e78fdeb97152e/providers/azure/aci.go#L115-L136